x/ibc/03-connection/handler.go

```diff
@@ -0,0 +1,126 @@
+package ics03
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/03-connection/keeper"
+	"github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+)
+
+// NewHandler creates a new Handler instance for IBC connection
+// transactions
+func NewHandler(k keeper.Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+
+		switch msg := msg.(type) {
+		case types.MsgConnectionOpenInit:
+			return handleMsgConnectionOpenInit(ctx, k, msg)
+
+		case types.MsgConnectionOpenTry:
+			return handleMsgConnectionOpenTry(ctx, k, msg)
+
+		case types.MsgConnectionOpenAck:
+			return handleMsgConnectionOpenAck(ctx, k, msg)
+
+		case types.MsgConnectionOpenConfirm:
+			return handleMsgConnectionOpenConfirm(ctx, k, msg)
+
+		default:
+			errMsg := fmt.Sprintf("unrecognized IBC connection message type: %T", msg)
+			return sdk.ErrUnknownRequest(errMsg).Result()
+		}
+	}
+}
+
+func handleMsgConnectionOpenInit(ctx sdk.Context, k keeper.Keeper, msg types.MsgConnectionOpenInit) sdk.Result {
+	err := k.ConnOpenInit(ctx, msg.ConnectionID, msg.ClientID, msg.Counterparty)
+	if err != nil {
+		return sdk.ResultFromError(err)
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeConnectionOpenInit,
+			sdk.NewAttribute(types.AttributeKeyConnectionID, msg.ConnectionID),
+			sdk.NewAttribute(types.AttributeKeyCounterpartyClientID, msg.Counterparty.ClientID),
```

Comment:
What about the client ID on the source chain?

Reply:
Nice suggestion. I'll add it